File: Java_Codes/10. Packages & Access Modifiers/EmployeeManager.java
import java.util.ArrayList;
import java.time.LocalDate;

// Employee class to store employee details
class Employee {
    private String name;
    private int age;
    private LocalDate joiningDate;

    public Employee(String name, int age, LocalDate joiningDate) {
        this.name = name;
        this.age = age;
        this.joiningDate = joiningDate;
    }

    // Method to return employee details as a formatted string
    public String getEmployeeDetails() {
        return name + ", " + age + ", Joined on: " + joiningDate;
    }
}

// Main class to manage employees
public class EmployeeManager {
    public static void main(String[] args) {
        // Using ArrayList to store employees
        ArrayList<Employee> employees = new ArrayList<>();
        employees.add(new Employee("Alice", 30, LocalDate.of(2020, 5, 20)));
        employees.add(new Employee("Bob", 25, LocalDate.of(2021, 8, 15)));
        employees.add(new Employee("Charlie", 28, LocalDate.of(2019, 3, 10)));

        // Displaying employee details
        System.out.println("Employee List:");
        for (Employee emp : employees) {
            System.out.println(emp.getEmployeeDetails());
        }
    }
}

File: Projects/PythonProject/Packages_And_Access_Modifiers/EmployeeManager.py
from datetime import date

# Employee class to store employee details
class Employee:
    def __init__(self, name, age, joining_date):
        self.name = name
        self.age = age
        self.joining_date = joining_date

    # Method to return employee details as a formatted string
    def get_employee_details(self):
        return f"{self.name}, {self.age}, Joined on: {self.joining_date}"

# Main logic to manage employees
def main():
    # Using a list to store employees
    employees = [
        Employee("Alice", 30, date(2020, 5, 20)),
        Employee("Bob", 25, date(2021, 8, 15)),
        Employee("Charlie", 28, date(2019, 3, 10))
    ]

    # Displaying employee details
    print("Employee List:")
    for emp in employees:
        print(emp.get_employee_details())

if __name__ == "__main__":
    main()
